src/gf-mksquashfs

```diff
@@ -0,0 +1,51 @@
+#!/usr/bin/env bash
+set -euo pipefail
+
+dn=$(dirname "$0")
+# shellcheck source=src/cmdlib.sh
+. "${dn}"/cmdlib.sh
+# shellcheck source=src/libguestfish.sh
+. "${dn}"/libguestfish.sh
+
+# Usage: gf-mksquashfs <input image> <output image>
+# Example: gf-mksquashfs fedora-coreos.qcow2 fedora-coreos.squashfs
+#
+# This will generate a squashfs from the contents of the root partition.
+
+src="$1"
+dest="$2"
+
+if [[ $src == *.gz || $src == *.xz ]]; then
+    img="$(basename "$src")"
+    fatal "Cannot generate squashfs from $img; not an uncompressed image"
+fi
+
+set -x
+# Work in a tmpdir on the destination so that we don't inherit some MCS labeling
+# from the /tmp dir in the container. This also ensures that the final move is a
+# pure `rename()`.
+# See also: https://github.com/coreos/coreos-assembler/issues/292
+tmpd=$(mktemp -tdp "$(dirname "${dest}")" gf-mksquashfs.XXXXXX)
+copy=${tmpd}/box.img
+tmp_dest=${tmpd}/image.squashfs
+
+# guestfish --ro appears to modify the disk image anyway.  Work on a copy.
```

Comment:
That'd be a pretty serious bug.   I assume you tried passing through `--ro` down into `coreos_gf_run_mount` somehow; are you sure it got passed correctly?

Testing this out quickly, it seems to work fine; I notice this in `/proc/$(pidof qemu)/fd`:
```
lr-x------. 1 walters walters 64 Sep  5 13:12 15 -> /var/srv/walters/fcos/builds/30.20190905.dev.1-1/x86_64/fedora-coreos-30.20190905.dev.1-1-qemu.qcow2
```

Reply:
(Of course, if we had O_OBJECT we'd also be setting it on our build artifacts too!)

Reply:
You're right; I misread `coreos_gf_launch`.